src/library/scala/concurrent/Future.scala

```diff
@@ -239,6 +239,17 @@ trait Future[+T] extends Awaitable[T] {
    */
   def map[S](f: T => S)(implicit executor: ExecutionContext): Future[S] = transform(_ map f)
 
+
+  /** Applies a side-effecting function to the encapsulated value in this Future
+    * @param f a function to apply to each element in this Future
+    * @tparam U the return type of f
+    * @return The same Future as this
+    */
+  def tapEach[U](f: T => U)(implicit executor: ExecutionContext): Future[T] = transform{ t =>
+    try t.foreach(f(_)) catch { case e if NonFatal(e) => executor.reportFailure(e)}
```

Comment:
I'm not sure this is the right behavior. I'd argue that if it throws an exception, the `Future` ought to become a failure

Reply:
Good point. I was mainly aiming to standardize the interface between different types i.e. `Option.tapEach` should behave as close as possible to `Future.tapEach` but I completely agree that the ability to contain the error within the Future is a powerful tool. I still want to run through a few example use cases but I will reiterate accordingly

Reply:
it might actually be easier to implement it for `Try`, and then just implement `Future#tapEach` as `transform(_ tapEach f)`